lib/defender/aws_iot_defender_collector.c

```diff
@@ -230,6 +230,7 @@ static void _serialize()
                                                         : assertSuccess;
 
     uint8_t metricsGroupCount = 0;
```

Comment:
I did not see this before... should never use anything other than a size_t (uint32_t) for an index. 8 bit quantityt does not really save anything, since CPU will have to load value in a register. And It may create stack disalignment...